amplify/backend/function/getKnowledgeGraph/src/index.js

```diff
@@ -39,59 +57,78 @@ exports.handler = async (event) => {
         id: paperInfo.data.paperId,
         label: paperInfo.data.title,
         type: "main",
+        referenceCount,
+        citationCount,
+        authors: getAuthorName(paperInfo.data),
+        published: paperInfo.data.publicationDate,
+        journal: getJournal(paperInfo.data),
+        venue: paperInfo.data.venue,
       },
     });
 
     // Adds a node for each referenced paper and sets up the
     // edges from current paper to all papers it references
     for (i = 1; i < referenceCount + 1; i++) {
       let reference = paperInfo.data.references[i - 1];
-      nodesAndEdges.push({
-        data: {
-          id: reference.paperId,
-          label: reference.title,
-          type: "reference",
-        },
-      });
-      nodesAndEdges.push({
-        data: {
-          source: paperInfo.data.paperId,
-          target: reference.paperId,
-          label: "Edge label placeholder",
-          type: "reference",
-        },
-      });
+      if (reference) {
+        nodesAndEdges.push({
+          data: {
+            id: reference.paperId,
+            label: reference.title,
+            type: "reference",
+            authors: getAuthorName(reference),
+            published: reference.publicationDate,
+            journal: getJournal(reference),
+            venue: reference.venue,
+          },
+        });
+        nodesAndEdges.push({
+          data: {
+            source: paperInfo.data.paperId,
+            target: reference.paperId,
+            label: "Edge label placeholder",
+            type: "reference",
+          },
+        });
+      }
     }
 
     // Adds a node for each citation and sets up the
     // edges from papers which cite this paper to the current paper
     for (i = referenceCount + 1; i < verticesCount; i++) {
       let citation = paperInfo.data.citations[i - referenceCount - 1];
-      nodesAndEdges.push({
-        data: {
-          id: citation.paperId,
-          label: citation.title,
-          type: "citation",
-        },
-      });
-      nodesAndEdges.push({
-        data: {
-          source: paperInfo.data.paperId,
-          target: citation.paperId,
-          label: "Edge label placeholder",
-          type: "citation",
-        },
-      });
+      if (citation) {
```

Comment:
Did you mean `citation.paperId` instead of just `citation`? Same goes for `reference` check above. I think we should check if the `paperId` is different from `null` cause it's still causing issues. For example with [this](https://arxiv.org/pdf/2211.14227.pdf) paper, if you look on the right side of the graph, you'll see some nodes without edges connecting them to the main node. If you search for those papers in the postman response for this paper, you'll see they have `paperId` set to `null`.
![image](https://user-images.githubusercontent.com/66720590/209444548-e0a7c78b-e088-4a62-aee5-09e14072dc4c.png
)The nodes saying "label edge placeholder" are also an issue but I'll recommend a fix for that in a separate comment.